Stellarium/stellarium
# Add plugin ExtraCalendar

### Description



Plugin `ExtraCalendar` displays multiple calendars (extensible, currently only support Chinese calendar) on the bottom-right corner. It helps user know how different calendars work with the influence of solar/lunar/etc.

Partially fixes #1313

Thanks to https://github.com/isee15/Lunar-Solar-Calendar-Converter/ 

### Screenshots (if appropriate):
![image](https://user-images.githubusercontent.com/69779107/96822670-2cbc8b00-145d-11eb-9ce8-7d8d17d657ce.png)
![image](https://user-images.githubusercontent.com/69779107/96822905-be2bfd00-145d-11eb-96f1-0af94fe95b58.png)
![image](https://user-images.githubusercontent.com/69779107/96822687-38a84d00-145d-11eb-82bb-90e9fc063342.png)
![image](https://user-images.githubusercontent.com/69779107/96822707-48c02c80-145d-11eb-9ed4-08765b130544.png)



### Type of change


### How Has This Been Tested?



- Test environment
  - macOS 10.15.7 Catalina
  - cmake 3.18.3
  - Qt 5.15.1
  - g++ Apple clang version 12.0.0 (clang-1200.0.32.2)
- Test steps
  - Make directory `build` under project root and enter into it
  - Run command `CMAKE_PREFIX_PATH=/usr/local/opt/qt cmake ..`
  - Run command `make -j4 && ./src/stellarium`
  - Open Configuration window, see if `ExtraCalendar` appears in the plugin tab
  - See if the description of`ExtraCalendar` plugin works as expected
  - See if `ExtraCalendar` plugin works as expected
  - Change default language of Stellarium, see if `ExtraCalendar` changes its name label as expected
  - Increase timespeed, see if `ExtraCalendar` updates correctly on local time 00:00AM
  - Press `+` and `-` to forward/backward dates, see if `ExtraCalendar` updates correctly
  - Press `F5`, change to date out of 1901~2099, see if `ExtraCalendar` disappears as expected
  - Go back to current time, see if `ExtraCalendar` shows again as expected
  - Change date to which is a Chinese leap month, see if `ExtraCalendar` works as expected
- Affected files
  - `/CMakeLists.txt`
  - `/po/stellarium/zh_CN.po`
  - `/src/core/StelApp.cpp`

**Test Configuration**:
* Operating system: macOS 10.15.7 Catalina
* Graphics Card: Intel Iris Plus Graphics 1536 MB

## Checklist:
